contracts/fixed_price_rev1.1.scilla

```diff
@@ -473,9 +536,28 @@ procedure HandleFulfillOrder(
     ZRC2TransferFrom payment_token_address buyer svc_fee_recipient svc_fee;
 
     (* On behalf of buyer, marketplace transfers the seller_profit to seller *)
-    ZRC2TransferFrom payment_token_address buyer seller seller_profit
+    ZRC2TransferFrom payment_token_address buyer seller seller_profit;
+
+    (* Paying out a commission if there is one *)
+    match is_commission_fee with
+      | True => (* No Operation *)
+      | False => 
+        ZRC2TransferFrom payment_token_address buyer brand_recipient brand_commission_fee 
+    end
   end;
 
+  msg_to_collection_contract = {
+    _tag: "TokenSaleCallback";
+    _recipient: collection_contract;  (* TODO: get this address from somewhere *)
```

Comment:
@sam-zilliqa shouldn't this be collections_contract or at least collection_address? If collection_contract is used here, it will be wrong once UpdateCollectionContract is called since it is collection_address that is updated. Also better to rename collection_contract to initial_collection_contract (same comment as in English auction).

Reply:
You are right, nicely spotted - but this value is fetched[ from this line](https://github.com/Zilliqa/marketplace-contracts/blob/3b62f7065b776cbd12129c1fcfd5a40e0bb063b6/contracts/fixed_price_rev1.1.scilla#L473) which always reads the collection_address, so even if the contract address is updated at some point, that update will be reflected here.